subprojects/cli/src/main/java/org/gradle/cli/CommandLineOption.java

```diff
@@ -61,8 +61,10 @@ public String getDescription() {
             if (result.length() > 0) {
                 result.append(' ');
             }
-            result.append("[deprecated - ");
-            result.append(deprecationWarning);
+            result.append("[deprecated");
+            if (!deprecationWarning.isEmpty()) {
+                result.append(" - ").append(deprecationWarning);
```

Comment:
:x: I'd personally vote for not adding the deprecation message. IMHO it doesn't serve any value when running `gradle -?`. It's really makes the output less readable. IMHO, rendering `deprecated` should be enough for a user to see that it should not be used anymore. I don't think "deprecated in version x" adds much value here. If we need a message here, then it should say "why we deprecated it and what the replacement is".

Reply:
Excellent, my opinion is same as you. I was hesitating if the messages should be removed.